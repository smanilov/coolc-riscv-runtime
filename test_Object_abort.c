typedef struct {
    int class_tag;
    int object_size;
    void* dispatch_table;
} Object;

extern Object* Object_abort();

int main() {
    Object_abort(); // expected: abort message

    return 0;
}
