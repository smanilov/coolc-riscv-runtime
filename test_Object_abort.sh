riscv64-elf-gcc -c -mabi=ilp32 -march=rv32imzicsr -nostdlib cc-rv-rt.s
sed "s/Object\.abort/Object_abort/g" -i cc-rv-rt.o 
riscv64-elf-gcc -mabi=ilp32 -march=rv32imzicsr -nostdlib -o a.out cc-rv-rt.o test_Object_abort.c -T cc-rv-rt.ld
spike --isa=RV32IMZICSR a.out > test_Object_abort.out

if diff -q test_Object_abort.out test_Object_abort.expected > /dev/null; then
    echo "PASSED"
else
    echo "FAILED"
fi
