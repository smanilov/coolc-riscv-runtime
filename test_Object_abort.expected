Program terminated due to call to Object_abort()
